src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/event/EventToSendProcessor.java

```diff
@@ -0,0 +1,65 @@
+package uk.gov.ons.ctp.integration.contactcentresvc.event;
+
+import static uk.gov.ons.ctp.common.log.ScopedStructuredArguments.kv;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Stream;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Propagation;
+import org.springframework.transaction.annotation.Transactional;
+import uk.gov.ons.ctp.common.domain.Source;
+import uk.gov.ons.ctp.common.event.EventPublisher;
+import uk.gov.ons.ctp.common.event.EventType;
+import uk.gov.ons.ctp.integration.contactcentresvc.config.AppConfig;
+import uk.gov.ons.ctp.integration.contactcentresvc.model.EventToSend;
+import uk.gov.ons.ctp.integration.contactcentresvc.repository.db.EventToSendRepository;
+
+@Slf4j
+@Component
+public class EventToSendProcessor {
+  private final EventToSendRepository eventToSendRepository;
+  private final EventPublisher eventPublisher;
+  private final AppConfig appConfig;
+
+  @Value("${scheduler.chunk-size}")
+  private int chunkSize;
+
+  public EventToSendProcessor(
+      EventToSendRepository repo, EventPublisher eventPublisher, AppConfig appConfig) {
+    this.eventToSendRepository = repo;
+    this.eventPublisher = eventPublisher;
+    this.appConfig = appConfig;
+  }
+
+  @Transactional(propagation = Propagation.REQUIRES_NEW) // Start a new transaction for every chunk
```

Comment:
there is no overlap because it is a fixed delay between calls

Reply:
Just a thought, even if they did overlap, would the SELECT FOR UPDATE prevent the same batch being processed the second time round.  I assume it would wait rather than pick another batch?

Reply:
Simon and I discussed and he is happy with the way it works.